Xamarin.Essentials/TextToSpeech/TextToSpeech.macos.cs

```diff
@@ -14,7 +14,8 @@ public static partial class TextToSpeech
 
         internal static Task<IEnumerable<Locale>> PlatformGetLocalesAsync() =>
             Task.FromResult(NSSpeechSynthesizer.AvailableVoices
-                .Select(v => new Locale(v, null, null, null)));
+                .Select(voice => NSSpeechSynthesizer.AttributesForVoice(voice))
+                .Select(attribute => new Locale(attribute["VoiceLanguage"].ToString(), null, attribute["VoiceName"].ToString(), attribute["VoiceIdentifier"].ToString())));
```

Comment:
Is there a chance for any of these to be null? Or the keys not exist?

Reply:
@mattleibow The docs (https://developer.apple.com/documentation/appkit/nsspeechsynthesizer/1448458-attributesforvoice?language=objc) don't mention anything, but you are right a `?` before the `ToString()` call wouldn't hurt and could stop some breakages.  Unless you have another suggestion for it?

Reply:
@mattleibow ? Checks added to latest commit.